### PR TITLE
Migrate CI from Travis to GitHub Action

### PR DESCRIPTION
**Pull request by mogeko**
Since June 15th, 2021, the building on [travis-ci.org](http://www.travis-ci.org/) is ceased.

So I tried to migrate CI to GitHub Action. It can be better integrated with GitHub, especially for Pull Request.

I mainly wrote these workflows:

- **build-test.yml**: Run `mix compile` and `mix test` in different versions of Elixir and OTP
- **benchmark.yml**: Run benchmark tasks. It can only be started manually.
- **release.yml**: After adding tags to git, synchronously publish to [hex.pm](https://hex.pm/packages/bento).

In addition, I need your help! 😃 @folz The workflow **release.yml** need add the `HEX_API_KEY` into the GitHub Action Secrets, It is only you can do this.

So, could you please help me to:

- Setup GitHub Action for this repository (if necessary) after you review the code I submitted.
- Generate a Key (with `Write` permissions) in our [hex organization](https://hex.pm/dashboard/orgs/bento), then add it to GitHub Action Secrets and name it to `HEX_API_KEY`.

Thank u!

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -58,15 +59,15 @@ defmodule Bento.Mixfile do
...
+      source_ref: "v#{@version}",
       extras: ["README.md", "LICENSE"]
     ]
   end
 
   defp package do
     [
       files: ~w(lib mix.exs README.md LICENSE VERSION),
-      maintainers: ["Rodney Folz"],
+      maintainers: ["Rodney Folz", "Zheng Junyi"],
```
Comment (folz) [inline mix.exs:70]: <3

**Review by folz — APPROVED** [review]
lgtm :shipit:

**Comment by folz** [comment]
@mogeko I added the API key to GH actions with that name. I'll let you merge this and verify it works, feel free to ping me if there's anything else I need to do.

**Comment by mogeko** [comment]
This is great! Thank you!

**Pull request merged**